tests/ui/consts/refs_check_const_eq-issue-88384.rs

```diff
@@ -8,10 +8,10 @@ struct CompileTimeSettings{
 }
 
 struct Foo<const T: CompileTimeSettings>;
-//~^ ERROR using function pointers as const generic parameters is forbidden
+//~^ ERROR `CompileTimeSettings` must be annotated with `#[derive(ConstParamTy)]` to be used as the type of a const generic parameter
```

Comment:
This is an example of the fixme I left above:

> // We should probably see if we're *allowed* to derive `ConstParamTy` on the type...

Since the user will follow the advice from the error message, then see an error message that `&'static [fn()]` is a bad field.